docker/lightgbm-ci-build-env-alpine/Dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM alpine:3.18.4
```

Comment:
What are the software versions on this alpine version? clang, cmake, etc? 

Lets use Zulu JDK which is the one being used by other builds.

Reply:
The software versions are:
  
  cmake=3.26.5
  make=4.4.1
  gcc  and g++: 12.2.1
  
  In the last commit I specified the version when running apk add. Also, clang was removed because it was not necessary.

Reply:
I also installed the same version of Zulu JDK that is being used on the other builds.